js/app/Data/DataManager.js

```diff
@@ -91,6 +91,13 @@ define(["app/Class", "app/Bounds", "lodash", "app/Events", "app/Data/Format", "a
       ));
     },
 
+    destroyView: function(view, sourceSpec) {
+      var self = this;
+      if (view.source && sourceSpec) {
+        self.removeSource(sourceSpec)
```

Comment:
This creates another bug - multiple animations (and therefore DataViews) can share one source.
We need to add reference-counting, just like for tiles.